pkg/machines/components/vm/firmwareModal.jsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import cockpit from 'cockpit';
+import { Button, Modal } from 'patternfly-react';
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import * as Select from "cockpit-components-select.jsx";
+import { setOSFirmware } from "../../libvirt-dbus.js";
+
+const _ = cockpit.gettext;
+
+export class FirmwareModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dialogError:  null,
+            firmware: props.firmware == "efi" ? _("UEFI") : _("BIOS"),
+        };
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+        this.close = props.close;
+        this.save = this.save.bind(this);
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    save() {
+        setOSFirmware(this.props.connectionName, this.props.vmId, this.state.firmware == _("UEFI") ? "efi" : "bios")
+                .then(this.close, exc => this.dialogErrorSet(_("Firmware change could not be saved"), exc.message));
+    }
+
+    render() {
+        return (
+            <Modal show onHide={this.close}>
+                <Modal.Header>
+                    <Modal.CloseButton onClick={this.close} />
+                    <Modal.Title> {_("Change Firmware")} </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <Select.Select
+                                   onChange={value => this.setState({ firmware: value })}
+                                   initial={this.state.firmware}
+                                   extraClass='form-control'>
+                        {[_("BIOS"), _("UEFI")].map(firmware => {
+                            return (
+                                <Select.SelectEntry data={firmware} key={firmware}>
+                                    { firmware }
+                                </Select.SelectEntry>
+                            );
+                        })}
```

Comment:
OK, but for two fixed entries only it'd be easier to just put two static `SelectEntry`s here. But either way, `key` and `data` should *not* be translated.

Reply:
Fixed both.